File: CgmLogger/src/main/java/com/kludgenics/cgmlogger/data/logdata/glucose/data/nightscout/MbgEntry.java
package com.kludgenics.cgmlogger.data.logdata.glucose.data.nightscout;

import com.google.gson.annotations.Expose;
import com.kludgenics.cgmlogger.data.logdata.glucose.data.BloodGlucose;
import io.realm.RealmObject;
import io.realm.annotations.Ignore;
import kotlin.Unit;

import java.util.Date;

/**
 * Created by matthiasgranberry on 5/24/15.
 */
public class MbgEntry implements BloodGlucose {
    @Expose
    private String device;
    @Expose
    private int mbg;
    @Expose
    private Date date;

    public MbgEntry() {
        super();
    }

    public MbgEntry(String device, Date date, int mbg) {
        this();
        this.device = device;
        this.mbg = mbg;
        this.date = date;
    }

    @Override
    public double getValue() {
        return mbg;
    }

    @Override
    public Date getDate() {
        return date;
    }

    @Override
    public String getType() {
        return BloodGlucose.TYPE_SMBG;
    }

    @Override
    public String getUnit() {
        return BloodGlucose.UNIT_MGDL;
    }
}

File: CgmLogger/src/main/java/com/kludgenics/cgmlogger/data/nightscout/MbgEntry.kt
package com.kludgenics.cgmlogger.data.nightscout

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName
import com.kludgenics.cgmlogger.data.glucose.BloodGlucose
import java.util.Date

/**
 * Created by matthiasgranberry on 5/24/15.
 */
public class MbgEntry() : BloodGlucose, NightscoutEntry {

    Expose
    SerializedName("_id")
    private var id: String = ""
    Expose
    private var device: String = ""
    Expose
    private var mbg: Int = 0
    Expose
    private var date: Date = Date()

    public constructor(device: String, date: Date, mbg: Int) : this() {
        this.device = device
        this.mbg = mbg
        this.date = date
    }

    override fun getValue(): Double {
        return mbg.toDouble()
    }

    override fun getId(): String {
        return id
    }

    override fun getDevice(): String {
        return device
    }

    override fun getDate(): Date {
        return date
    }

    override fun getType(): String {
        return BloodGlucose.TYPE_SMBG
    }

    override fun getUnit(): String {
        return BloodGlucose.UNIT_MGDL
    }
}